include/net/mqtt_socket.h

```diff
@@ -0,0 +1,655 @@
+	 */
+	MQTT_EVT_CONNACK,
+
+	/** Disconnection Event. MQTT Client Reference is no longer valid once
+	 *  this event is received for the client.
+	 */
+	MQTT_EVT_DISCONNECT,
+
+	/** Publish event received when message is published on a topic client
+	 *  is subscribed to.
+	 */
+	MQTT_EVT_PUBLISH,
+
+	/** Acknowledgment for published message with QoS 1. */
+	MQTT_EVT_PUBACK,
+
+	/** Reception confirmation for published message with QoS 2. */
+	MQTT_EVT_PUBREC,
+
+	/** Release of published published messages with QoS 2. */
+	MQTT_EVT_PUBREL,
+
+	/** Confirmation to a publish release message. Applicable only to
+	 *  QoS 2 messages.
+	 */
+	MQTT_EVT_PUBCOMP,
+
+	/** Acknowledgment to a subscribe request. */
+	MQTT_EVT_SUBACK,
+
+	/** Acknowledgment to a unsubscribe request. */
+	MQTT_EVT_UNSUBACK
+} mqtt_evt_id_t;
+
+/**@brief MQTT version protocol level. */
+typedef enum {
+	MQTT_VERSION_3_1_0      = 3,    /**< Protocol level for 3.1.0. */
+	MQTT_VERSION_3_1_1      = 4     /**< Protocol level for 3.1.1. */
+} mqtt_version_t;
+
+/**@brief MQTT Quality of Service types. */
+typedef enum {
+	/** Lowest Quality of Service, no acknowledgment needed for published
+	 *  message.
+	 */
+	MQTT_QoS_0_AT_MOST_ONCE = 0x00,
+
+	/** Medium Quality of Service, if acknowledgment expected for published
+	 *  message, duplicate messages permitted.
+	 */
+	MQTT_QoS_1_ATLEAST_ONCE = 0x01,
+
+	/** Highest Quality of Service, acknowledgment expected and message
+	 *  shall be published only once. Message not published to interested
+	 *  parties unless client issues a PUBREL.
+	 */
+	MQTT_QoS_2_EACTLY_ONCE  = 0x02
+} mqtt_qos_t;
+
+/**@brief MQTT CONNACK return codes. */
+typedef enum {
+	/** Connection accepted. */
+	MQTT_CONNECTION_ACCEPTED                = 0x00,
+
+	/** The Server does not support the level of the MQTT protocol
+	 * requested by the Client.
+	 */
+	MQTT_UNACCEPTABLE_PROTOCOL_VERSION      = 0x01,
+
+	/** The Client identifier is correct UTF-8 but not allowed by the
+	 *  Server.
+	 */
+	MQTT_IDENTIFIER_REJECTED                = 0x02,
+
+	/** The Network Connection has been made but the MQTT service is
+	 *  unavailable.
+	 */
+	MQTT_SERVER_UNAVAILABLE                 = 0x03,
+
+	/** The data in the user name or password is malformed. */
+	MQTT_BAD_USER_NAME_OR_PASSWORD          = 0x04,
+
+	/** The Client is not authorized to connect. */
+	MQTT_NOT_AUTHORIZED                     = 0x05
+} mqtt_conn_return_code_t;
+
+/**@brief MQTT SUBACK return codes. */
+typedef enum {
+	/** Subscription with QoS 0 succeeded. */
+	MQTT_SUBACK_SUCCESS_QoS_0 = 0x00,
+
+	/** Subscription with QoS 1 succeeded. */
+	MQTT_SUBACK_SUCCESS_QoS_1 = 0x01,
+
+	/** Subscription with QoS 2 succeeded. */
+	MQTT_SUBACK_SUCCESS_QoS_2 = 0x02,
+
+	/** Subscription for a topic failed. */
+	MQTT_SUBACK_FAILURE = 0x80
+} mqtt_suback_return_code_t;
+
+/**@brief MQTT client forward declaration @ref mqtt_client_t for details. */
+typedef struct mqtt_client_t mqtt_client_t;
```

Comment:
We typically do not use typedef for structs and enums in zephyr.

Reply:
I wasn't sure if it's a requirement to avoid `typedef` in Zephyr, so I've left it as it was originally. Of course I can change that as it's requested.